src/Appwrite/Auth/Auth.php

```diff
@@ -130,26 +144,56 @@ public static function hash(string $string)
      *
      * One way string hashing for user passwords
      *
-     * @param $string
+     * @param string $string
+     * @param string $algo hashing algorithm to use
+     * @param string $options algo-specific options
      *
      * @return bool|string|null
      */
-    public static function passwordHash($string)
+    public static function passwordHash(string $string, string $algo, mixed $options = [])
     {
-        return \password_hash($string, PASSWORD_BCRYPT, array('cost' => 8));
+        // Plain text not supported, just an alias. Switch to recommended algo
+        if ($algo === 'plaintext') {
+            $algo = Auth::DEFAULT_ALGO;
+            $options = Auth::DEFAULT_ALGO_OPTIONS;
+        }
+
+        if (!\array_key_exists($algo, Auth::SUPPORTED_ALGOS)) {
+            throw new \Exception('Hashing algorithm \'' . $algo . '\' is not supported.');
+        }
+
+        $className = Auth::SUPPORTED_ALGOS[$algo];
+        $classPath = '\\Appwrite\\Auth\\Hash\\' . $className;
+        $hasher = new $classPath($options);
```

Comment:
Not a huge fan of this. I know we do it when we create an OAuth adapter. Would a switch case work here?
Also, having a custom, well defined signature for initing each hash would be much friendlier.